### PR TITLE
feat: Add shared layer to read data from sharing endpoint

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

feat: Add shared layer to read data from sharing endpoint

Fix https://github.com/datafuselabs/databend/issues/7821

### CODE REVIEW

**Comment by Xuanwo** [comment]
Cc @ZhiHanZ to confirm the sharing API.

**Review by lichuang — APPROVED** [review]

File: src/query/sharing/src/signer.rs [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+impl Debug for SharedSinger {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("SharedSinger")
+            .field("endpoint", &self.endpoint)
+            .finish_non_exhaustive()
+    }
+}
+
+impl SharedSinger {
```
Comment (ZhiHanZ) [inline src/query/sharing/src/signer.rs:53]: lint: SharedSinger -> SharedSigner

File: src/query/sharing/src/signer.rs [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        self.sign_inner(reqs).await
+    }
+
+    /// Databend Cloud Presign API will be:
+    ///
+    /// ```shell
+    /// curl -X POST \
+    ///      https://endpoint/tenant/<tenant_id>/database/<db_id>/table/<table_id>/presign \
+    ///      --header 'Authorization Bearer OIDC TOKEN' \
+    ///      --data-raw '[{"path": "file_a", "method": "GET"}, {"path": "file_b", "method": "PUT"}'
```
Comment (ZhiHanZ) [inline src/query/sharing/src/signer.rs:113]: I think this design make sense, as we could add additonal fields such as customized headers, expiration time in the PresignRequestItem with backward compatibility

**Review by ZhiHanZ — APPROVED** [review]
LGTM, we could add additonal e2e tests after PoC complete

**Pull request merged**